docs/read-the-docs/source/workceptor.rst

```diff
@@ -0,0 +1,182 @@
+Workceptor
+==========
+
+Workceptor is a component of receptor that handles units of work.
+
+``work-command`` defines a type of work that can run on the node.
+
+foo.yml
+
+.. code-block:: yaml
+
+    ---
+    - node:
+        id: foo
+
+    - log-level:
+        level: Debug
+
+    - tcp-listener:
+        port: 2222
+
+    - control-service:
+        service: control
+        filename: /tmp/foo.sock
+
+    - work-command:
+        workType: echosleep
+        command: bash
+        params:  "-c \"for i in {1..5}; do echo $i; done\""
+
+bar.yml
+
+.. code-block:: yaml
+
+    ---
+    - node:
+        id: bar
+
+    - log-level:
+        level: Debug
+
+    - tcp-peer:
+        address: localhost:2222
+
+    - control-service:
+        service: control
+
+    - work-command:
+        worktype: echosleep
+        command: bash
+        params:  "-c \"for i in {1..10}; do echo $i; sleep 1; done\""
+
+
+Configuring work commands
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``worktype`` User-defined name to give this work definition
+
+``command`` The executable that is invoked when running this work
+
+``params`` Command-line options passed to this executable
+
+
+Local work
+^^^^^^^^^^
+
+Start the work by connecting to the ``control-service`` and issuing a "work submit" command
+
+.. code-block::
+
+    $ receptorctl --socket /tmp/foo.sock work submit echosleep --no-payload
+    Result:  Job Started
+    Unit ID: t1BlAB18
+
+Receptor started an instance of this work type, and labeled it with a unique "Unit ID"
+
+Work results
+^^^^^^^^^^^^
+
+Use the "Unit ID" to get work results
+
+.. code-block::
+
+    receptorctl --socket /tmp/foo.sock work results t1BlAB18
+    1
+    2
+    3
+    4
+    5
+    6
+    7
+    8
+    9
+    10
+
+
+Remote work
+^^^^^^^^^^^
+Although connected to `foo`, by providing the "--node" option the work can be started on node `bar`.
+
+Note: the work type must be defined on the node it is intended to run on, e.g. `bar` must have a ``work-command`` called "echosleep", in this case.
+
+.. code-block::
+
+    $ receptorctl --socket /tmp/foo.sock work submit echosleep --node bar --no-payload
+    Result:  Job Started
+    Unit ID: 87Vwqb6A
+
+Sequence of events for remote work submission
+
+- `foo` starts a local work unit of work type "remote". This is a special work type that is built into receptor.
+- This work unit attempts to connect to `bar`'s control service and issue a "work submit echosleep" command. Note, from `bar`'s perspective, this is the exact same operation as if a user connected to `bar` directly and issued a work submit command. `bar` is not aware that `foo` is the one that issued the command.
+- Once submitted, `foo` will stream work results back to `foo` and store it on disk. It also periodically gets the ``status`` of the work running on `bar`. Status includes information about the work state and the stdout size.
```

Comment:
> Once submitted, `foo` will stream work results back to `foo`

`bar` will stream work results back to `foo`?

Reply:
or is this saying that `foo` will request that work results be streamed back to itself?

Reply:
yes the latter, maybe `foo will stream work results back to itself` is more clear